pandas/core/arrays/datetimes.py

```diff
@@ -829,6 +829,24 @@ def to_period(self, freq=None):
 
         return PeriodArrayMixin(self.values, freq=freq)
 
+    def to_perioddelta(self, freq):
+        """
+        Calculate TimedeltaArray of difference between index
+        values and index converted to PeriodArray at specified
+        freq. Used for vectorized offsets
```

Comment:
I find this a rather "obscure" method (meaning, not directly sure how it would be useful for me as a user). But I am also not a frequent user of periods/offsets, so difficult to say (and maybe a good example in the docstring might already help). 

So what I want to say: in case this is indeed rather for internal use, I would not make it a public method on the array class (but can still do the move but make it for example private here)

Reply:
This is used indirectly in DatetimeIndex + DateOffset, so will be needed for DatetimeArray + DateOffset.